src/main/java/github/scarsz/discordsrv/util/PlaceholderUtil.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * DiscordSRV - A Minecraft to Discord and back link plugin
+ * Copyright (C) 2016-2019 Austin "Scarsz" Shapiro
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package github.scarsz.discordsrv.util;
+
+import github.scarsz.discordsrv.DiscordSRV;
+import github.scarsz.discordsrv.hooks.VaultHook;
+import github.scarsz.discordsrv.objects.Lag;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+// A bunch of general placeholders.
+public class PlaceholderUtil {
+    // Tries to apply placeholderAPI placeholders
+    public static String applyPlaceholderApi(Player player, String text) {
+        if (PluginUtil.pluginHookIsEnabled("placeholderapi")) return me.clip.placeholderapi.PlaceholderAPI.setPlaceholders(player, text);
```

Comment:
While you're at it, PAPI translates colors; thus we should translate them back (& -> §)

Reply:
PlaceholderAPI is used for discord as well however, so we'd get a lot of unnecessary colour codes in discord

Reply:
what i'm saying is for Discord messages; role mentions break because of PAPI because it's converting <@&id> to <@§id>

Reply:
Oh I see. How could we work around that though? Only thing I can think of is replacing the `&` with some other unicode character that people are unlikely to use, then replacing it back after PAPI does it's thing.

Or maybe a placeholder like %role_ROLEID% that subs it out for a proper role mention after all the other placeholders are done...that makes more sense.